snuba/utils/streams/batching.py

```diff
@@ -305,12 +305,11 @@ def signal_shutdown(self) -> None:
         self.__shutdown_requested = True
 
     def _shutdown(self) -> None:
-        if self.__processor is not None:
-            logger.debug("Stopping processor")
-            self.__processor.close()
-            self.__processor = None
-
         # close the consumer
         logger.debug("Stopping consumer")
         self.__consumer.close()
         logger.debug("Stopped")
+
+        # if there was an active processor, it should be shut down and unset
+        # when the partitions are revoked during consumer close
+        assert self.__processor is None, "processor was not closed on shutdown"
```

Comment:
See manual tests here, though fixing the behavior of `DummyConsumer.close` also manifested this behavior: https://github.com/getsentry/snuba/pull/1139#issuecomment-659596608